eogrow/utils/filter.py

```diff
@@ -30,19 +31,32 @@ def check_if_features_exist(
     return len(not_seen_features) == 0
 
 
-def get_patches_without_all_features(
+def get_patches_with_missing_features(
     filesystem: FS,
     patches_folder: str,
     patch_list: List[str],
     features: Iterable[Feature],
 ) -> List[str]:
-    """Filters out patches that already contain all features"""
+    """Filters out names of those EOPatches that are missing some given features.
+
+    :param filesystem: A filesystem object.
+    :param patches_folder: A path to folder with EOPatches, relative to `filesystem` object.
+    :param patch_list: A list of EOPatch names.
+    :param features: A list of EOPatch features.
+    :return: A sublist of `patch_list` with only EOPatch names that are missing some features.
+    """
     eopatch_paths = [fs.path.combine(patches_folder, eopatch) for eopatch in patch_list]
 
     def check_patch(eopatch_path: str) -> bool:
         return check_if_features_exist(filesystem, eopatch_path, features)
 
-    with ThreadPoolExecutor() as pool:
-        has_features_iter = pool.map(check_patch, eopatch_paths)
+    with ThreadPoolExecutor() as executor:
+        has_features_list = list(
+            tqdm(
```

Comment:
is any of our tqdm + threads solutions directly applicable here?

Reply:
For now I didn't want to complicate with the "proper" solutions as we have in `eo-learn` and `sentinelhub-py` where progress bar correctly measures complete checks with `as_completed`. That is mainly because in this case checking for each EOPatch should take approximately the same amount of time and checking starts from the start of the list.

But in the future I hope that we generalize parallelization utilities (either in `eolearn.core` or `eogrow.utils.parallelize`) where we properly handle progress bar and then we can use those for this use case as well.

Reply:
hmmmm, sure